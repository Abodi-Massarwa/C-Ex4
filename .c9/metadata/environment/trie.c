{"filter":false,"title":"trie.c","tooltip":"/trie.c","undoManager":{"mark":37,"position":37,"stack":[[{"start":{"row":191,"column":56},"end":{"row":192,"column":0},"action":"insert","lines":["",""],"id":3},{"start":{"row":192,"column":0},"end":{"row":192,"column":4},"action":"insert","lines":["    "]}],[{"start":{"row":192,"column":4},"end":{"row":192,"column":5},"action":"insert","lines":["c"],"id":4},{"start":{"row":192,"column":5},"end":{"row":192,"column":6},"action":"insert","lines":["h"]},{"start":{"row":192,"column":6},"end":{"row":192,"column":7},"action":"insert","lines":["a"]}],[{"start":{"row":192,"column":6},"end":{"row":192,"column":7},"action":"remove","lines":["a"],"id":5},{"start":{"row":192,"column":5},"end":{"row":192,"column":6},"action":"remove","lines":["h"]},{"start":{"row":192,"column":4},"end":{"row":192,"column":5},"action":"remove","lines":["c"]}],[{"start":{"row":192,"column":4},"end":{"row":192,"column":5},"action":"insert","lines":["c"],"id":6},{"start":{"row":192,"column":5},"end":{"row":192,"column":6},"action":"insert","lines":["h"]},{"start":{"row":192,"column":6},"end":{"row":192,"column":7},"action":"insert","lines":["a"]},{"start":{"row":192,"column":7},"end":{"row":192,"column":8},"action":"insert","lines":["r"]},{"start":{"row":192,"column":8},"end":{"row":192,"column":9},"action":"insert","lines":["*"]}],[{"start":{"row":192,"column":9},"end":{"row":192,"column":10},"action":"insert","lines":[" "],"id":7},{"start":{"row":192,"column":10},"end":{"row":192,"column":11},"action":"insert","lines":["p"]},{"start":{"row":192,"column":11},"end":{"row":192,"column":12},"action":"insert","lines":["a"]},{"start":{"row":192,"column":12},"end":{"row":192,"column":13},"action":"insert","lines":["t"]},{"start":{"row":192,"column":13},"end":{"row":192,"column":14},"action":"insert","lines":["h"]},{"start":{"row":192,"column":14},"end":{"row":192,"column":15},"action":"insert","lines":["n"]},{"start":{"row":192,"column":15},"end":{"row":192,"column":16},"action":"insert","lines":["a"]},{"start":{"row":192,"column":16},"end":{"row":192,"column":17},"action":"insert","lines":["m"]},{"start":{"row":192,"column":17},"end":{"row":192,"column":18},"action":"insert","lines":["e"]}],[{"start":{"row":192,"column":18},"end":{"row":192,"column":19},"action":"insert","lines":[";"],"id":8}],[{"start":{"row":192,"column":4},"end":{"row":192,"column":19},"action":"remove","lines":["char* pathname;"],"id":9},{"start":{"row":192,"column":4},"end":{"row":192,"column":5},"action":"insert","lines":["c"]},{"start":{"row":192,"column":5},"end":{"row":192,"column":6},"action":"insert","lines":["h"]},{"start":{"row":192,"column":6},"end":{"row":192,"column":7},"action":"insert","lines":["a"]},{"start":{"row":192,"column":7},"end":{"row":192,"column":8},"action":"insert","lines":["r"]}],[{"start":{"row":192,"column":8},"end":{"row":192,"column":10},"action":"insert","lines":["[]"],"id":10}],[{"start":{"row":192,"column":8},"end":{"row":192,"column":10},"action":"remove","lines":["[]"],"id":11}],[{"start":{"row":192,"column":8},"end":{"row":192,"column":9},"action":"insert","lines":[" "],"id":12},{"start":{"row":192,"column":9},"end":{"row":192,"column":10},"action":"insert","lines":["p"]},{"start":{"row":192,"column":10},"end":{"row":192,"column":11},"action":"insert","lines":["a"]},{"start":{"row":192,"column":11},"end":{"row":192,"column":12},"action":"insert","lines":["t"]},{"start":{"row":192,"column":12},"end":{"row":192,"column":13},"action":"insert","lines":["h"]},{"start":{"row":192,"column":13},"end":{"row":192,"column":14},"action":"insert","lines":["n"]},{"start":{"row":192,"column":14},"end":{"row":192,"column":15},"action":"insert","lines":["a"]},{"start":{"row":192,"column":15},"end":{"row":192,"column":16},"action":"insert","lines":["m"]},{"start":{"row":192,"column":16},"end":{"row":192,"column":17},"action":"insert","lines":["e"]}],[{"start":{"row":192,"column":17},"end":{"row":192,"column":19},"action":"insert","lines":["[]"],"id":13}],[{"start":{"row":192,"column":18},"end":{"row":192,"column":19},"action":"insert","lines":["2"],"id":14},{"start":{"row":192,"column":19},"end":{"row":192,"column":20},"action":"insert","lines":["6"]}],[{"start":{"row":192,"column":21},"end":{"row":192,"column":22},"action":"insert","lines":[";"],"id":15}],[{"start":{"row":192,"column":22},"end":{"row":193,"column":0},"action":"insert","lines":["",""],"id":16},{"start":{"row":193,"column":0},"end":{"row":193,"column":4},"action":"insert","lines":["    "]}],[{"start":{"row":200,"column":15},"end":{"row":200,"column":23},"action":"remove","lines":["filename"],"id":17},{"start":{"row":200,"column":15},"end":{"row":200,"column":16},"action":"insert","lines":["p"]},{"start":{"row":200,"column":16},"end":{"row":200,"column":17},"action":"insert","lines":["a"]},{"start":{"row":200,"column":17},"end":{"row":200,"column":18},"action":"insert","lines":["t"]},{"start":{"row":200,"column":18},"end":{"row":200,"column":19},"action":"insert","lines":["h"]}],[{"start":{"row":200,"column":15},"end":{"row":200,"column":19},"action":"remove","lines":["path"],"id":18},{"start":{"row":200,"column":15},"end":{"row":200,"column":23},"action":"insert","lines":["pathname"]}],[{"start":{"row":193,"column":4},"end":{"row":193,"column":5},"action":"insert","lines":["s"],"id":19},{"start":{"row":193,"column":5},"end":{"row":193,"column":6},"action":"insert","lines":["t"]}],[{"start":{"row":193,"column":5},"end":{"row":193,"column":6},"action":"remove","lines":["t"],"id":20},{"start":{"row":193,"column":4},"end":{"row":193,"column":5},"action":"remove","lines":["s"]}],[{"start":{"row":193,"column":4},"end":{"row":193,"column":5},"action":"insert","lines":["g"],"id":21},{"start":{"row":193,"column":5},"end":{"row":193,"column":6},"action":"insert","lines":["e"]},{"start":{"row":193,"column":6},"end":{"row":193,"column":7},"action":"insert","lines":["t"]}],[{"start":{"row":193,"column":6},"end":{"row":193,"column":7},"action":"remove","lines":["t"],"id":22},{"start":{"row":193,"column":5},"end":{"row":193,"column":6},"action":"remove","lines":["e"]},{"start":{"row":193,"column":4},"end":{"row":193,"column":5},"action":"remove","lines":["g"]}],[{"start":{"row":193,"column":4},"end":{"row":193,"column":5},"action":"insert","lines":["g"],"id":23},{"start":{"row":193,"column":5},"end":{"row":193,"column":6},"action":"insert","lines":["e"]},{"start":{"row":193,"column":6},"end":{"row":193,"column":7},"action":"insert","lines":["t"]},{"start":{"row":193,"column":7},"end":{"row":193,"column":8},"action":"insert","lines":["c"]}],[{"start":{"row":193,"column":8},"end":{"row":193,"column":9},"action":"insert","lines":["w"],"id":24},{"start":{"row":193,"column":9},"end":{"row":193,"column":10},"action":"insert","lines":["d"]}],[{"start":{"row":193,"column":10},"end":{"row":193,"column":12},"action":"insert","lines":["()"],"id":25}],[{"start":{"row":193,"column":11},"end":{"row":193,"column":12},"action":"insert","lines":["p"],"id":26},{"start":{"row":193,"column":12},"end":{"row":193,"column":13},"action":"insert","lines":["a"]},{"start":{"row":193,"column":13},"end":{"row":193,"column":14},"action":"insert","lines":["t"]},{"start":{"row":193,"column":14},"end":{"row":193,"column":15},"action":"insert","lines":["h"]}],[{"start":{"row":193,"column":11},"end":{"row":193,"column":15},"action":"remove","lines":["path"],"id":27},{"start":{"row":193,"column":11},"end":{"row":193,"column":19},"action":"insert","lines":["pathname"]}],[{"start":{"row":193,"column":19},"end":{"row":193,"column":20},"action":"insert","lines":[","],"id":28}],[{"start":{"row":193,"column":20},"end":{"row":193,"column":21},"action":"insert","lines":["s"],"id":29},{"start":{"row":193,"column":21},"end":{"row":193,"column":22},"action":"insert","lines":["t"]},{"start":{"row":193,"column":22},"end":{"row":193,"column":23},"action":"insert","lines":["r"]},{"start":{"row":193,"column":23},"end":{"row":193,"column":24},"action":"insert","lines":["l"]}],[{"start":{"row":193,"column":23},"end":{"row":193,"column":24},"action":"remove","lines":["l"],"id":30},{"start":{"row":193,"column":22},"end":{"row":193,"column":23},"action":"remove","lines":["r"]},{"start":{"row":193,"column":21},"end":{"row":193,"column":22},"action":"remove","lines":["t"]},{"start":{"row":193,"column":20},"end":{"row":193,"column":21},"action":"remove","lines":["s"]}],[{"start":{"row":193,"column":20},"end":{"row":193,"column":21},"action":"insert","lines":["s"],"id":31},{"start":{"row":193,"column":21},"end":{"row":193,"column":22},"action":"insert","lines":["t"]},{"start":{"row":193,"column":22},"end":{"row":193,"column":23},"action":"insert","lines":["r"]},{"start":{"row":193,"column":23},"end":{"row":193,"column":24},"action":"insert","lines":["l"]},{"start":{"row":193,"column":24},"end":{"row":193,"column":25},"action":"insert","lines":["e"]}],[{"start":{"row":193,"column":20},"end":{"row":193,"column":25},"action":"remove","lines":["strle"],"id":32},{"start":{"row":193,"column":20},"end":{"row":193,"column":26},"action":"insert","lines":["strlen"]}],[{"start":{"row":193,"column":26},"end":{"row":193,"column":28},"action":"insert","lines":["()"],"id":33}],[{"start":{"row":193,"column":27},"end":{"row":193,"column":28},"action":"insert","lines":["p"],"id":34},{"start":{"row":193,"column":28},"end":{"row":193,"column":29},"action":"insert","lines":["a"]},{"start":{"row":193,"column":29},"end":{"row":193,"column":30},"action":"insert","lines":["t"]},{"start":{"row":193,"column":30},"end":{"row":193,"column":31},"action":"insert","lines":["h"]}],[{"start":{"row":193,"column":27},"end":{"row":193,"column":31},"action":"remove","lines":["path"],"id":35},{"start":{"row":193,"column":27},"end":{"row":193,"column":35},"action":"insert","lines":["pathname"]}],[{"start":{"row":193,"column":20},"end":{"row":193,"column":36},"action":"remove","lines":["strlen(pathname)"],"id":36},{"start":{"row":193,"column":20},"end":{"row":193,"column":56},"action":"insert","lines":["sizeof(pathname)/sizeof(pathname[0])"]}],[{"start":{"row":193,"column":57},"end":{"row":193,"column":58},"action":"insert","lines":[";"],"id":37}],[{"start":{"row":193,"column":58},"end":{"row":194,"column":0},"action":"insert","lines":["",""],"id":38},{"start":{"row":194,"column":0},"end":{"row":194,"column":4},"action":"insert","lines":["    "]}],[{"start":{"row":194,"column":4},"end":{"row":194,"column":31},"action":"insert","lines":["strcat(pathname, filename);"],"id":39}],[{"start":{"row":191,"column":20},"end":{"row":191,"column":44},"action":"remove","lines":["/home/ubuntu/environment"],"id":40}]]},"ace":{"folds":[],"scrolltop":0,"scrollleft":0,"selection":{"start":{"row":197,"column":24},"end":{"row":197,"column":24},"isBackwards":false},"options":{"guessTabSize":true,"useWrapMode":false,"wrapToView":true},"firstLineState":0},"timestamp":1579186716223,"hash":"e63cb49cf28436063511b22686e21182bdc97f72"}