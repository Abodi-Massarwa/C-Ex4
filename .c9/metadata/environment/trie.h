{"filter":false,"title":"trie.h","tooltip":"/trie.h","undoManager":{"mark":-1,"position":-1,"stack":[]},"ace":{"folds":[],"scrolltop":634.5,"scrollleft":0,"selection":{"start":{"row":0,"column":0},"end":{"row":0,"column":0},"isBackwards":false},"options":{"guessTabSize":true,"useWrapMode":false,"wrapToView":true},"firstLineState":{"row":38,"state":"start","mode":"ace/mode/c_cpp"}},"timestamp":1579183003471,"hash":"7ae5ebb7744ffaa890dd01a7d5be8e7fcb948056"}