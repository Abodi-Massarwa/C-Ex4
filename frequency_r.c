//
// Created by Abodi Msarwi on 15/01/2020.
//
#include <stdio.h>
#include "trie.h"
#define CHARS_SIZE 26
#define LINE_SIZE 256
#define FALSE 0
#define TRUE 1
extern String* WORDS;
extern int iterator;
int main()
{
    Node* root=NULL;
    root= newnode(root);
    Trie* tree=NULL;
    newtrie(&root,&tree);
    while(get_line(root)!=FALSE){}
    print_words_reverse(root,iterator);
    return 0;
}

