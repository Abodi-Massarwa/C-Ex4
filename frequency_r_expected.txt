worst   1
wisdom  1
was 4
times   2
the 4
of  4
it  4
foolishness 1
best    1
age 2
